src/project_types/rust.rs

```diff
@@ -25,32 +25,65 @@ pub fn find_cargo_tomls(repo: &Repository) -> Result<Vec<VersionFile>> {
                 continue;
             }
 
-            let toml: toml_edit::Document = read_to_string(path)
+            let toml: toml_edit::DocumentMut = read_to_string(path)
                 .with_context(|| format!("Failed reading file {path:?} "))?
                 .parse()
-                .context("Failed parsing Cargo.toml file")?;
+                .with_context(|| format!("Failed parsing {path:?}"))?;
 
-            if let Some(version) = toml["package"]["version"]
-                .as_str()
+            if let Some(version) = toml
+                .get("package")
+                .and_then(|t| t.get("version")?.as_str())
                 .map(Version::parse)
                 .transpose()
-                .with_context(|| format!("Failed parsing version for {:?}", path))?
+                .with_context(|| format!("Failed parsing version for {path:?}"))?
             {
-                debug!("Found Cargo.toml: {:?} (version={})", path, version);
-                returned.push(VersionFile::CargoToml(path.to_owned(), version))
+                debug!("Found Cargo.toml: {path:?} (version={version})");
+                returned.push(VersionFile::CargoToml {
+                    path: path.to_owned(),
+                    version,
+                    is_workspace: false,
+                })
+            } else if let Some(version) = toml
```

Comment:
What is the difference between this condition and the one on line #33 (the green one)?

Reply:
This one checks `workspace.package.version` and the previous one checks `package.version`